src/instructlab/sdg/utils/taxonomy.py

```diff
@@ -108,37 +110,55 @@ def _get_documents(
     skip_checkout: bool = False,
 ) -> List[str]:
     """
-    Retrieve the content of files from a Git repository.
+    Retrieve the content of files (Markdown and PDF) from a Git repository.
 
     Args:
         source (dict): Source info containing repository URL, commit hash, and list of file patterns.
+        skip_checkout (bool, optional): If True, skips checking out the specific commit. Defaults to False.
 
     Returns:
-         List[str]: List of document contents.
-    """ ""
+        List[str]: List of document contents (Markdown as text and PDFs as extracted text).
+
+    Raises:
+        SystemExit: If no valid documents are found.
+        OSError, GitCommandError, FileNotFoundError: For errors during Git operations or file access.
+    """
     repo_url = source.get("repo")
     commit_hash = source.get("commit")
     file_patterns = source.get("patterns", [])
-    with tempfile.TemporaryDirectory() as temp_dir:
-        try:
-            repo = git.Repo.clone_from(repo_url, temp_dir)
-            if not skip_checkout:
-                repo.git.checkout(commit_hash)
-
-            file_contents = []
 
-            logger.debug("Processing files...")
-            for pattern in file_patterns:
-                for file_path in glob.glob(os.path.join(repo.working_dir, pattern)):
-                    if os.path.isfile(file_path) and file_path.endswith(".md"):
+    try:
+        repo = git.Repo.clone_from(repo_url, DOC_FILEPATH)
+        if not skip_checkout:
+            repo.git.checkout(commit_hash)
+
+        file_contents = []
+
+        logger.debug("Processing files...")
+        for pattern in file_patterns:
+            for file_path in glob.glob(os.path.join(repo.working_dir, pattern)):
+                if os.path.isfile(file_path):
+                    if file_path.endswith(".md"):
+                        # Process markdown files
                         with open(file_path, "r", encoding="utf-8") as file:
                             file_contents.append(file.read())
+                    elif file_path.endswith(".pdf"):
+                        # Process PDF files
+                        with open(file_path, "rb") as file:
+                            reader = PdfReader(file)  # Use pypdf PdfReader
+                            pdf_text = ""
+                            for page in reader.pages:  # Iterating through pages
+                                pdf_text += page.extract_text()
+                            file_contents.append(pdf_text)
+
+        if file_contents:
+            filepaths = [DOC_FILEPATH / p for p in file_patterns]
```

Comment:
Prefer `os.path.join` to combine paths to string concatenation - `os.path.join` handles various edge cases like ensuring we don't add double slashes and more complex paths.